irr/include/irrArray.h

```diff
@@ -45,6 +45,9 @@ class array
 	{
 	}
 
+	//! Move constructor
+	array(std::vector<T> &&data) : m_data(data), is_sorted(false) {}
```

Comment:
Thanks, fixed in https://github.com/minetest/minetest/pull/14557/commits/7a9b58b85f8dc7fd6fc77adf27965bbeaa003c4f

Reply:
I probably shouldn't have suggested std::forward down there, as we don't have a forwarding reference here, sorry.
Using `std::forward` with an explicit type (e.g. `std::forward<std::vector<u16>>`) is probably a bad idea. It will always move. (But it looks like it did magic, which is deceiving.) You'd normally only want to pass template type parameters or similar in the angled brackets, or let it be auto deducted.
See here: https://godbolt.org/z/vMfKM5sP5

Reply:
I see. Thanks for the correction. Done in https://github.com/minetest/minetest/pull/14557/commits/75acbad42824b6ceaaac620fc85294240735b58f.